clients/src/main/java/org/apache/kafka/common/requests/FetchResponse.java

```diff
@@ -73,19 +74,19 @@
     private final int throttleTime;
 
     public static final class PartitionData {
-        public final short errorCode;
+        public final Errors error;
         public final long highWatermark;
         public final Records records;
 
-        public PartitionData(short errorCode, long highWatermark, Records records) {
-            this.errorCode = errorCode;
+        public PartitionData(Errors error, long highWatermark, Records records) {
+            this.error = error;
             this.highWatermark = highWatermark;
             this.records = records;
         }
 
         @Override
         public String toString() {
-            return "(errorCode=" + errorCode + ", highWaterMark=" + highWatermark +
+            return "(errorCode=" + error.code() + ", highWaterMark=" + highWatermark +
```

Comment:
nit: maybe it's more helpful to use the error directly here since no one knows the codes.